channel/channel_compress.go

```
@@ -59,6 +68,96 @@ func detectEncoder() (videoEncoder, string) {
 	return availableEncoders[len(availableEncoders)-1], "CPU"
 }
 
+// getFpsPassthroughFlag returns the ffmpeg flag(s) that pass each video
+// frame's timestamp straight through to the muxer. Modern ffmpeg uses
+// -fps_mode passthrough; older 4.x exits with "Unrecognized option
+// 'fps_mode'", so we probe support once and fall back to -vsync passthrough
+// (deprecated on 5+, but still functional through ffmpeg 8).
+func getFpsPassthroughFlag() []string {
+	fpsPassthroughOnce.Do(func() {
+		cmd := exec.Command("ffmpeg", "-hide_banner", "-loglevel", "error",
```

golangci-lint (2.12.2)

[error] os/exec.Command must not be called. use os/exec.CommandContext

(noctx)